### ISSUE TITLE
HttpRepl as Powershell Provider

### ISSUE THREAD

**Issue by Platzer** [dotnet/HttpRepl#188]
Hi everyone,

I'm just watching the ASP.NET Community Standup from August 27th and thought that it would be a good fit if this tool would be provided as a [Powershell Provider](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_providers?view=powershell-6) because:
- there would be command history out of the box
- infrastructure for auto completion and other command line actions would be there (e.g. `ctrl + space` showing all possible commands)
- it would be possible to save the commands as `.ps1` scripts and reuse them
- and the best thing it could be integrated with all the available powershell commands (e.g. `Where-Object` or `Select-Object` for filtering and reducing the response)

BR, Lukas

**Comment by tlmii** [comment]
Thanks @Platzer. Good to know this would be of interest to you. We've gotten this feedback a couple times now in the last few days and its definitely something we want to explore.

**Comment by bradygaster** [comment]
We talked about this and agree it is a legitimate use-case and scenario. However, it is something we agreed was orthogonal to `httprepl`, so we're closing this issue. 

Under the supervision of @beefarino some time ago I dabbled in the creation of a Powershell Provider. We basically built a prototype of a provider that sat atop Fluent NHibernate-backed data sources (oh the pain and the joy when it worked). I've mentioned Jim here in hopes that he will be inspired to run off and create a universal Open API-consuming Powershell Provider. :) 

I frankly love the idea of an Open API Powershell Provider, and would love to see something happen with it in the community. But given the goals of the `httprepl` project are to meet .NET Core developers where they are - the .NET CLI - this feels like it's a different project altogether.
